src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -92,7 +92,14 @@ public final Response send(
         final InputStream content, final int connect, final int read
     ) throws IOException {
         final Response rsp;
-        if (method.equals(Request.GET)) {
+        boolean requestHasCmch = false;
```

Comment:
Perhaps we should only bother calculating this when it's a GET request since this is relatively expensive.